stubs/toml/toml/encoder.pyi

```diff
@@ -0,0 +1,30 @@
+from _typeshed import SupportsWrite
+from typing import Any, Callable, Generic, Iterable, Mapping, MutableMapping, TypeVar
+
+_MappingT = TypeVar("_MappingT", bound=Mapping[str, Any])
+
+def dump(o: _MappingT, f: SupportsWrite[str], encoder: TomlEncoder[_MappingT] | None = ...) -> str: ...
+def dumps(o: _MappingT, encoder: TomlEncoder[_MappingT] | None = ...) -> str: ...
+
+class TomlEncoder(Generic[_MappingT]):
+    _dict: type[_MappingT]
+    preserve: bool
+    dump_funcs: MutableMapping[type[Any], Callable[[Any], str]]
+    def __init__(self, _dict: type[_MappingT] = ..., preserve: bool = ...) -> None: ...
```

Comment:
This too needs overloads to specify which mapping type to use when `_dict` is not given. `TomlPreserveInlineDictEncoder` and `TomlArraySeparatorEncoder` need to be updated too.